cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -57,18 +66,36 @@
    */
   @Override
   public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreException, SQLException {
-    Location namespaceHome = namespacedLocationFactory.get(namespaceMeta.getNamespaceId().toId());
-    if (namespaceHome.exists()) {
-      LOG.warn("Home directory '{}' for namespace '{}' already exists. Deleting it.",
-               namespaceHome, namespaceMeta.getName());
-      if (!namespaceHome.delete(true)) {
-        throw new IOException(String.format("Error while deleting home directory '%s' for namespace '%s'",
-                                            namespaceHome, namespaceMeta.getName()));
+    Location namespaceHome;
+    String rootDirectory = namespaceMeta.getConfig().getRootDirectory();
+    if (Strings.isNullOrEmpty(rootDirectory)) {
```

Comment:
see https://github.com/caskdata/cdap/pull/6068/files#r71074638. This class is soon going to be unmanageable. We should move the FS stuff to its own private create/delete methods at the very least.